python/tests-integration/test_signer.py

```diff
@@ -0,0 +1,140 @@
+        f.write(_GPG_INSTRUCTIONS)
+    genkey = subprocess.run(
+        [
+            "gpg",
+            "--batch",
+            "--generate-key",
+            "--pinentry-mode",
+            "loopback",
+            f"{dir.name}/instructions",
+        ],
+        input=os.devnull,
+        capture_output=True,
+        check=False,
+        text=True,
+        env={"GNUPGHOME": dir.name},
+    )
+    if genkey.returncode != 0:
+        print(genkey.stderr)
+        raise pytest.fail()
+
+    # Save the public key
+    pubkey = subprocess.run(
+        ["gpg", "--export", "--armor"],
+        capture_output=True,
+        check=False,
+        text=True,
+        env={"GNUPGHOME": dir.name},
+    )
+    if pubkey.returncode != 0:
+        print(pubkey.stderr)
+        raise pytest.fail()
+    with open(f"{dir.name}/public.gpg", "w") as f:
+        f.write(pubkey.stdout)
+
+    # Save the private key
+    privkey = subprocess.run(
+        [
+            "gpg",
+            "--export-secret-keys",
+            "--pinentry-mode",
+            "loopback",
+            "--yes",
+            "--armor",
+        ],
+        capture_output=True,
+        check=False,
+        text=True,
+        env={"GNUPGHOME": dir.name},
+    )
+    if privkey.returncode != 0:
+        print(privkey.stderr)
+        raise pytest.fail()
+    with open(f"{dir.name}/private.gpg", "w") as f:
+        f.write(privkey.stdout)
+
+    yield f"{dir.name}/private.gpg", f"{dir.name}/public.gpg"
+
+    del dir
+
+
+@pytest.mark.parametrize(
+    "playbook",
+    [
+        # official production playbooks
+        "playbooks/insights_remove.yml",
+        # custom playbooks signed by official Red Hat key
+        "playbooks/document-from-hell.yml",
+        # unsigned playbooks
+        "playbooks-unsigned/sample.yml",
+    ],
+)
+def test_end_to_end(ephemeral_gpg_keys: tuple[str, str], playbook: str):
+    """Test that we can sign and verify a playbook."""
+    signing_result = subprocess.run(
+        [
+            "insights-ansible-playbook-signer",
+            "--playbook",
+            f"{DATA_DIRECTORY / playbook}",
+            "--key",
+            ephemeral_gpg_keys[0],
+            "--debug",
+        ],
+        capture_output=True,
+        text=True,
+        check=False,
+        env={**os.environ, "LC_ALL": "C.UTF-8"},
+    )
+    if signing_result.returncode != 0:
+        print(signing_result.stderr.strip())
+    assert signing_result.returncode == 0
+
+    reading_result = subprocess.run(
+        [
+            "insights-ansible-playbook-verifier",
+            "--stdin",
+            "--key",
+            ephemeral_gpg_keys[1],
+            "--skip-revocation-list",
+            "--debug",
+        ],
+        input=signing_result.stdout,
+        capture_output=True,
+        text=True,
+        check=False,
+        env={**os.environ, "LC_ALL": "C.UTF-8"},
+    )
+    if reading_result.returncode != 0:
+        print(reading_result.stderr.strip())
```

Comment:
ditto

and missing the check for `returncode == 0`?

Reply:
That's strictly not necessary, but won't hurt.